caom2/caom2/observation.py

```diff
@@ -318,22 +318,23 @@ def sequence_number(self):
 
     @sequence_number.setter
     def sequence_number(self, value):
-        caom_util.type_check(value, int, 'sequence_number')
-        self._sequence_number = value
+        caom_util.type_check(value, int_32, 'sequence_number')
+        self._sequence_number = int_32(value) if value is not None else None
+
 
     @property
-    def obs_type(self):
+    def type(self):
         """The OBSTYPE of the observation being recorded.
 
         eg. OBJECT, FLAT, BIAS
         type: unicode str
         """
-        return self._obs_type
+        return self._type
 
-    @obs_type.setter
-    def obs_type(self, value):
-        caom_util.type_check(value, str, 'obs_type')
-        self._obs_type = value
+    @type.setter
+    def type(self, value):
```

Comment:
Attributes in the model are accessed with @property type getters/setters. get_* are utility methods that return something based on the existing metadata (such as an URI based on a few components). That being said, I removed set_uri from the observation as setters are not allowed in that case.